compilers/concrete-compiler/compiler/lib/Conversion/SimulateTFHE/SimulateTFHE.cpp

```diff
@@ -92,6 +92,88 @@ struct NegOpPattern : public mlir::OpConversionPattern<TFHE::NegGLWEOp> {
   }
 };
 
+mlir::Value globalStringValueFromLoc(mlir::ConversionPatternRewriter &rewriter,
+                                     mlir::Location loc) {
+  std::string locString;
+  auto ros = llvm::raw_string_ostream(locString);
+  loc.print(ros);
+
+  std::string msgName;
+  std::stringstream stream;
+  stream << "loc_" << rand();
```

Comment:
Should not be very problematic, but names can conflicts (I'm ok it's unlikely) and above all make compilation not repeatable so It could be preferable to use a counter here.